cloud_governance/common/timeout/timeout_retry.py

```diff
@@ -0,0 +1,47 @@
+from functools import wraps
+import time
+
+from requests import ReadTimeout
+
+from cloud_governance.common.logger.init_logger import logger
+
+RETRIES = 5
+SLEEP_TIME = 30
+
+
+def timeout_retry(method):
+    """
+    This method call to wrap method
+    @param method:
+    @return: method wrapper
+    """
+    @wraps(method)  # solve method help doc
+    def method_wrapper(*args, **kwargs):
+        """
+        This method wrap the input method
+        @param args:
+        @param kwargs:
+        @return: prefix + input method + suffix
+        """
+        retires_count = 0
+        result = []
+        waiting_time = 0
+        while RETRIES != retires_count:
+            try:
+                result = method(*args, **kwargs)
+            except ReadTimeout as err:
+                waiting_time += SLEEP_TIME
+                logger.info(f'Server did not respond, retry after {SLEEP_TIME}. Total waiting time {waiting_time}')
+                time.sleep(SLEEP_TIME)
+                retires_count += 1
+            except Exception as err:
+                logger.info(err)
+                break
+            if result:
+                logger.info(f'Respond time {waiting_time}')
+                return result
+            retires_count += 0
+        else:
+            logger.info('Server not responding, quit the process without doing further retires')
+        return result
+    return method_wrapper
```

Comment:
There is a existing retry decorator: https://pypi.org/project/retry/

Reply:
ok, Let me take a look at it.